### DISCUSSION TITLE
Which data types are ImPlot users using?

### DISCUSSION THREAD

**Discussion by epezent** [epezent/implot#328] [Q&A]
ImPlot currently supports 10 data types for all plot types -- `float, double, ImS8, ImU8, ImS16, ImU16, ImS32, ImU32, ImS64, ImU64` . I suspect most folks have data that is of type `float`, `double`, or `int` (and maybe an unsigned integral). So, I'm curious if anyone out there is using any of the other 5 on a regular basis. 

PS -- Don't worry, I'm not planning removal. Just considering placing the lesser used ones behind a `#ifdef IMPLOT_SUPPORT_ALL_TYPES` to cut down on compile time and size.

**Comment by marcizhu** [comment]
I personally just use `float`/`double` and in some rare cases I use `int` for plotting histograms and stuff like that. I bet most users use the same data types I use (just like you mentioned).
